src/app/static/src/app/components/plots/choropleth/Choropleth.vue

```diff
@@ -0,0 +1,160 @@
+        const currentFeatures = ref<Feature[]>([]);
+        const featureData = ref<IndicatorValuesDict>({});
+
+        const map = ref<typeof LMap | null>(null);
+        const featureRefs = ref<typeof LGeoJson[]>([]);
+
+        const {createTooltips, updateTooltips} = useMapTooltips(featureData, indicatorMetadata, currentFeatures, featureRefs)
+
+        const updateMap = () => {
+            updateFeatures();
+            updateColourScales();
+            updateTooltips();
+        };
+        // Watch on the plotData only instead of using computed. Here we want to update the colours,
+        // the features and the legend. If we use computed this will cause multiple updates
+        // 1. When a selection changes it will update immediately
+        // 2. After the plotData has been fetched async it will update again
+        // This can cause the map to appear to flicker as it updates with old scales
+        // then quickly updates after new data has been fetched.
+        // Instead manually watch on the plot data changes, and also trigger this when a user changes the
+        // scale selection
+        watch([plotData], updateMap)
+
+        const updateColourScales = () => {
+            const colourScales = store.state.plotState.output.colourScales;
+            const selectedIndicator =  store.getters["plotSelections/selectedIndicator"];
+            selectedScale.value = colourScales[selectedIndicator];
+            if (!selectedScale.value) {
+                selectedScale.value = initialiseScaleFromMetadata(indicatorMetadata.value);
+                updateOutputColourScale(selectedScale.value);
+            }
+            colourRange.value = getColourRange(indicatorMetadata.value, selectedScale.value, plotData.value);
+            scaleLevels.value = getScaleLevels(indicatorMetadata.value, colourRange.value);
+            featureData.value = getFeatureData(
+                    plotData.value,
+                    indicatorMetadata.value,
+                    colourRange.value ? colourRange.value : {max: 1, min: 0}
+            );
+        }
+
+        const updateFeatures = () => {
+            indicatorMetadata.value = store.getters["modelCalibrate/indicatorMetadata"];
+            const selectedLevel = store.state.plotSelections.choropleth.filters
+                    .find(f => f.stateFilterId === "detail")!.selection;
+            const selectedAreas = store.state.plotSelections.choropleth.filters
+                    .find(f => f.stateFilterId === "area")!.selection;
+            currentFeatures.value = getVisibleFeatures(features, selectedLevel, selectedAreas);
+        }
+
+        const updateBounds = () => {
+            if (featureRefs.value.length > 0) {
+                map.value?.leafletObject.fitBounds(featureRefs.value.map(f => f.leafletObject.getBounds()));
+            }
+        };
+        watch(featureRefs.value, () => {
+            updateBounds();
+        });
```

Comment:
Maybe move this up next to the other watch to keep those together.

Reply:
I tried this but it causes weird behaviour with the bounds updating. I think if we put the `updateBounds` into the earlier watcher it runs before the `featureRefs` have been updated. You can see this if you 
1. Open the map
2. Select "district" area level
3. Deselect "Northern"
4. Switch been deselect and selecting "Central", you will see the map bounds update the previously selected set of features instead of the current one

So I think we need this in a separate watcher which is run after those featureRefs update. 

There is still a bit of stuttering with this in this PR but I have fixed this in the next one.

Reply:
I didn't really mean combine them into the same watch, I literally just meant to put them next to each other in the file to make the code a bit easier to navigate/understand! :)